Magento2/Sniffs/Exceptions/DirectThrowSniff.php

```diff
@@ -44,7 +44,29 @@ public function process(File $phpcsFile, $stackPtr)
         $tokens = $phpcsFile->getTokens();
         $endOfStatement = $phpcsFile->findEndOfStatement($stackPtr);
         $posOfException = $phpcsFile->findNext(T_STRING, $stackPtr, $endOfStatement);
-        if ($tokens[$posOfException]['content'] === 'Exception') {
+
+        $fullExceptionString = $this->getFullClassNameAndAlias($tokens, $stackPtr, $endOfStatement);
+        $exceptionString = 'Exception';
+        $customExceptionFound = false;
+        foreach ($tokens as $key => $token) {
+            if ($token['code'] === T_USE) {
```

Comment:
I would write it differently, to avoid excessive nesting and make it easier to read
```suggestion
           if ($token['code'] !== T_USE) {
                continue;
           }
           $endOfUse = $phpcsFile->findEndOfStatement($key);
           $useStatementValue = $this->getFullClassNameAndAlias($tokens, $key, $endOfUse);
           // we safely consider use statement has alias will not be a direct exception class
           if (!empty($useStatementValue['alias'])) {
                continue;
           }
           if (substr($useStatementValue['name'], 0, strlen($exceptionString)) !== $exceptionString
                && substr($useStatementValue['name'], -strlen($exceptionString)) === $exceptionString
                && $useStatementValue['name'] !== $exceptionString
                ) {
                        $customExceptionFound = true;
                        break;
                }
           }
```